cmd/kogito/command/deploy/deploy_service.go

```diff
@@ -278,3 +319,149 @@ func (i *deployCommand) Exec(cmd *cobra.Command, args []string) (err error) {
+	// if found download it and save on /tmp
+	switch {
+	case strings.HasPrefix(resource, "http"):
+		parsedURL, err := url.ParseRequestURI(resource)
+		if err != nil {
+			return nil, nil, fmt.Errorf("kogito resource is not a valid URL, received %s", resource)
+		}
+
+		hasExtension := false
+		ff := strings.Split(parsedURL.Path, "/")
+		fileName = strings.Join(strings.Fields(ff[len(ff)-1]), "")
+
+		if isSuffixSupported(fileName) {
+			hasExtension = true
+		}
+
+		if hasExtension {
+			response, err := http.Get(resource)
+			if err != nil {
+				return nil, nil, fmt.Errorf("failed to download %s, error message: %s", resource, err.Error())
+			}
+			options.AsFile = fileName
+			return response.Body, options, nil
+		}
+
+	// handle single file, files on dir.
+	default:
+		fileInfo, err := os.Stat(resource)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		if fileInfo.Mode().IsRegular() {
+			if isSuffixSupported(resource) {
+				log.Infof("File found %s.", resource)
+				ff := strings.Split(resource, "/")
+				fileName = strings.Join(strings.Fields(ff[len(ff)-1]), "")
+				fileReader, err := os.Open(resource)
+				if err != nil {
+					return nil, nil, err
+				}
+				options.AsFile = fileName
+				return fileReader, options, nil
+			}
+		}
+
+		if fileInfo.Mode().IsDir() {
+			log.Info("The provided source is a dir, packing files.")
+			if err != nil {
+				return nil, nil, err
+			}
+
+			var buf bytes.Buffer
+			gzipWriter := gzip.NewWriter(&buf)
+			tarWriter := tar.NewWriter(gzipWriter)
+
+			var link string
+			filepath.Walk(resource, func(absoluteFilePath string, fileInfo os.FileInfo, err error) error {
+				if isSuffixSupported(fileInfo.Name()) {
+
+					if fileInfo.Mode()&os.ModeSymlink != 0 {
+						link, err = os.Readlink(absoluteFilePath)
+						if err != nil {
+							return err
+						}
+					}
+					header, err := tar.FileInfoHeader(fileInfo, link)
+					if err != nil {
+						return err
+					}
+					header.Name = filepath.ToSlash(fileInfo.Name())
+					header.Linkname = filepath.ToSlash(header.Linkname)
+					header.Format = tar.FormatPAX
+
+					// write header
+					if err := tarWriter.WriteHeader(header); err != nil {
+						return err
+					}
+
+					fileToCompress, err := os.Open(absoluteFilePath)
+					if err != nil {
+						return err
+					}
+					defer fileToCompress.Close()
+
+					if _, err := io.Copy(tarWriter, fileToCompress); err != nil {
+						return err
+					}
+				}
+
+				return nil
+			})
+
+			// produce tar
+			if err := tarWriter.Close(); err != nil {
+				return nil, nil, err
+			}
+			// produce gzip
+			if err := gzipWriter.Close(); err != nil {
+				return nil, nil, err
+			}
+			tarWriter.Flush()
+
+			var r io.Reader
+			br := bufio.NewReaderSize(&buf, 4096)
+			r = br
+			fileToWrite, err1 := os.OpenFile(filepath.Join("/tmp", "compressed_kogito_resources.tgz"), os.O_CREATE|os.O_RDWR, os.FileMode(777))
```

Comment:
remind for later, use tmp dir from go env.

Reply:
can you please apply the change in this PR so it won't be forgotten?
```suggestion
			fileToWrite, err1 := ioutil.TempFile("", "compressed_kogito_resources_*.tgz")
```